doc/introduction/compiling_circuits.rst

```diff
@@ -0,0 +1,305 @@
+
+
+>>> with qml.transforms.set_decomposition(custom_decomps, original_dev):
+...     print(qml.draw(original_qnode, expansion_strategy="device")(weights))
+0: ──RX(0.4)──────────────────────╭C──RZ(3.14)──RY(1.57)──────────────────────────╭Z──RZ(3.14)──RY(1.57)──┤ ⟨Z⟩
+1: ──RX(0.5)──RZ(3.14)──RY(1.57)──╰Z──RZ(3.14)──RY(1.57)──╭C──────────────────────│───────────────────────┤
+2: ──RX(0.6)──RZ(3.14)──RY(1.57)──────────────────────────╰Z──RZ(3.14)──RY(1.57)──╰C──────────────────────┤
+
+Circuit cutting
+---------------
+
+Circuit cutting allows you to replace a circuit with ``N`` wires by a set of circuits with less than
+``N`` wires (see also `Peng et. al <https://arxiv.org/abs/1904.00102>`_). Of course this comes with a cost: The smaller circuits
+require a greater number of device executions to be evaluated.
+
+In PennyLane, circuit cutting can be
+activated by positioning :class:`~.pennylane.WireCut` operators at the desired cut locations, and
+by decorating the QNode with the :func:`~.pennylane.transforms.cut_circuit` transform.
+
+The example below shows how a three-wire circuit can be run on a two-wire device:
+
+.. code-block:: python
+
+    dev = qml.device("default.qubit", wires=2)
+
+    @qml.cut_circuit
+    @qml.qnode(dev)
+    def circuit(x):
+        qml.RX(x, wires=0)
+        qml.RY(0.9, wires=1)
+        qml.RX(0.3, wires=2)
+
+        qml.CZ(wires=[0, 1])
+        qml.RY(-0.4, wires=0)
+
+        qml.WireCut(wires=1)
+
+        qml.CZ(wires=[1, 2])
+
+        return qml.expval(qml.grouping.string_to_pauli_word("ZZZ"))
+
+Instead of executing the circuit directly, it will be partitioned into
+smaller fragments according to the :class:`~.pennylane.WireCut` locations,
+and each fragment executed multiple times. PennyLane automatically combines the results
+of the fragment executions to recover the expected output of the original uncut circuit.
+
+>>> x = np.array(0.531, requires_grad=True)
+>>> circuit(0.531)
+0.47165198882111165
+
+Circuit cutting support is also differentiable:
+
+>>> qml.grad(circuit)(x)
+-0.276982865449393
+
+Simulated quantum circuits that produce samples can be cut using
+the :func:`~.pennylane.transforms.cut_circuit_mc`
+transform based on the Monte Carlo method:
+
+.. code-block:: python
+
+    dev = qml.device("default.qubit", wires=2, shots=1000)
+
+    @qml.cut_circuit_mc
+    @qml.qnode(dev)
+    def circuit(x):
+        qml.RX(0.89, wires=0)
+        qml.RY(0.5, wires=1)
+        qml.RX(1.3, wires=2)
+
+        qml.CNOT(wires=[0, 1])
+        qml.WireCut(wires=1)
+        qml.CNOT(wires=[1, 2])
+
+        qml.RX(x, wires=0)
+        qml.RY(0.7, wires=1)
+        qml.RX(2.3, wires=2)
+        return qml.sample(wires=[0, 2])
+
+>>> x = 0.3
+>>> circuit(x)
+tensor([[1, 1],
+        [0, 1],
+        [0, 1],
+        ...,
+        [0, 1],
+        [0, 1],
+        [0, 1]], requires_grad=True)
+
+The samples are drawn from the same distribution that the original
+circuit gives rise to.
```

Comment:
unrelated, but I wonder if it makes sense to have a section for transforms that _change_ circuit outputs/structure that aren't necessarily compilation? e.g.,

- `insert`
- `mitigate_with_zne`

I also need to find a place (probably not here) for `metric_tensor` and `adjoint_metric_tensor`

Reply:
Let's add later, this is maybe just a start? 

Once this is in master we could ask people to check out the new structure and add important features as the teams see fit...

Reply:
Makes sense! The reason `insert` was on my mind is that it came up during the papers project that people didn't realize it existed.